q2_feature_classifier/_cutter.py

```diff
@@ -98,75 +103,92 @@ def _approx_match(seq, f_primer, r_primer, identity):
     return None
 
 
-def _gen_reads(sequences,  f_primer, r_primer, trunc_len, trim_left, identity,
+def _gen_reads(sequence, f_primer, r_primer, trunc_len, trim_left, identity,
                min_length, max_length):
     f_primer = skbio.DNA(f_primer)
     r_primer = skbio.DNA(r_primer)
-    for seq in sequences:
-        amp = _exact_match(seq, f_primer, r_primer)
-        if not amp:
-            amp = _exact_match(seq.reverse_complement(), f_primer, r_primer)
-        if not amp:
-            amp = _approx_match(seq, f_primer, r_primer, identity)
-        if not amp:
-            amp = _approx_match(
-                seq.reverse_complement(), f_primer, r_primer, identity)
-        if not amp:
-            continue
-        # we want to filter by max length before trimming
-        if max_length > 0 and len(amp) > max_length:
-            continue
-        if trunc_len > 0:
-            amp = amp[:trunc_len]
-        if trim_left > 0:
-            amp = amp[trim_left:]
-        if min_length > 0 and len(amp) < min_length:
-            continue
-        if not amp:
-            continue
-        yield amp
-
-
-def extract_reads(sequences: DNAIterator,  f_primer: str, r_primer: str,
-                  trunc_len: int = 0, trim_left: int = 0,
+    amp = _exact_match(sequence, f_primer, r_primer)
+    if not amp:
+        amp = _exact_match(sequence.reverse_complement(), f_primer, r_primer)
+    if not amp:
+        amp = _approx_match(sequence, f_primer, r_primer, identity)
+    if not amp:
+        amp = _approx_match(
+            sequence.reverse_complement(), f_primer, r_primer, identity)
+    if not amp:
+        return
+    # we want to filter by max length before trimming
+    if max_length > 0 and len(amp) > max_length:
+        return
+    if trunc_len > 0:
+        amp = amp[:trunc_len]
+    if trim_left > 0:
+        amp = amp[trim_left:]
+    if min_length > 0 and len(amp) < min_length:
+        return
+    if not amp:
+        return
+    return amp
+
+
+def extract_reads(sequences: DNASequencesDirectoryFormat,  f_primer: str,
+                  r_primer: str, trunc_len: int = 0, trim_left: int = 0,
                   identity: float = 0.8, min_length: int = 50,
-                  max_length: int = 0) -> DNAIterator:
+                  max_length: int = 0, n_jobs: int = 1,
+                  batch_size: int = 'auto') -> DNAFASTAFormat:
```

Comment:
The type annotation doesn't matter here. Personally, I prefer to set this type to whatever the default value's type is, but, its six of one, half-dozen of the other.